pkg/osquery/extension.go

```diff
@@ -786,6 +797,21 @@ func (e *Extension) writeResultsWithReenroll(ctx context.Context, results []dist
 }
 
 func getEnrollDetails(client Querier) (service.EnrollmentDetails, error) {
+	var details service.EnrollmentDetails
+
+	// To facilitate manual testing around missing enrollment details,
+	// there is a environmental variable to trigger the failure condition
+	if os.Getenv("LAUNCHER_DEBUG_ENROLL_DETAILS_ERROR") == "true" {
+		return details, errors.New("Skipping enrollment details")
+	}
+
+	// This condition is indicative of a misordering (or race) in
+	// startup. Enrollment has started before `SetQuerier` has
+	// been called.
+	if client == nil {
```

Comment:
I'm mostly just making sure I'm following things correctly, but...

given https://github.com/kolide/launcher/pull/819/files#diff-222645cc15c22456115e733d8ff807db8ea12a5839e79aff41a0159afa21abfcR122, whenever the transport is 'osquery', we're going to have a nil client (because we return before calling `setQuerier`), but that is fine because if the transport is osquery we aren't ever going to be calling getEnrollmentDetails, correct?

(I see what you mean by this being all very tangled...)

Reply:
Well, this _is_ the extension, so if we're using osquery transport, then this is the thing we're not starting. It should never be called. 

But, yes, then the `client == nil` case would be there. though the real purpose is because something I did this week got a panic because this was called before that Querier was set. I think I've since undone that but it seems good to leave it in.